pallets/parachain-staking/src/migrations.rs

```diff
@@ -36,8 +36,95 @@ use frame_support::{
 };
 use sp_std::{collections::btree_map::BTreeMap, vec::Vec};
 
+/// TODO: migration from CandidateState to CandidateInfo
+pub struct SplitCandidateStateToDecreasePoV<T>(PhantomData<T>);
+impl<T: Config> OnRuntimeUpgrade for IncreaseMaxDelegationsPerCandidate<T> {
+	fn on_runtime_upgrade() -> Weight {
+		let (mut reads, mut writes) = (0u64, 0u64);
+		for (account, state) in <CandidateState<T>>::iter() {
+			reads += 1u64;
+			// 1. collect all delegations into single vec and order them
+			let mut all_delegations = state.top_delegations.clone();
```

Comment:
Initializing this vec with the size it will later contain (after appending `starting_bottom_delegations` below) might have a big impact on performance since it can avoid copying later.